nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/AccessResource.java

```diff
@@ -329,24 +359,221 @@ public Response oidcExchange(@Context HttpServletRequest httpServletRequest, @Co
+                                    .build();
+                            HttpPost httpPost = new HttpPost(revokeEndpoint);
+
+                            List<NameValuePair> params = new ArrayList<>();
+                            params.add(new BasicNameValuePair("token", accessToken));
+                            httpPost.setEntity(new UrlEncodedFormEntity(params));
+
+                            try (CloseableHttpResponse response = httpClient.execute(httpPost)) {
+                                httpClient.close();
+
+                                if (response.getStatusLine().getStatusCode() == HTTPResponse.SC_OK) {
+                                    // Redirect to logout page
+                                    logger.debug("You are logged out of the OpenId Connect Provider.");
+                                    String postLogoutRedirectUri = generateResourceUri("..", "nifi", "logout-complete");
+                                    httpServletResponse.sendRedirect(postLogoutRedirectUri);
+                                } else {
+                                    logger.error("There was an error logging out of the OpenId Connect Provider. " +
+                                            "Response status: " + response.getStatusLine().getStatusCode());
+                                }
+                            }
+
+                        } catch (final IOException e) {
+                            logger.error("There was an error logging out of the OpenId Connect Provider: "
+                                    + e.getMessage(), e);
+
+                            // Remove the oidc request cookie
+                            removeOidcRequestCookie(httpServletResponse);
+
+                            // Forward to the error page
+                            forwardToMessagePage(httpServletRequest, httpServletResponse,
+                                    "There was an error logging out of the OpenId Connect Provider: "
+                                            + e.getMessage());
+                        }
+                    }
+                    break;
+                case ID_TOKEN_LOGOUT:
+                    // Use the end session endpoint + ID Token
+                    final String idToken;
+                    try {
+                        // Return the ID Token
+                        idToken = oidcService.exchangeAuthorizationCodeForIdToken(authorizationGrant);
+                    } catch (final Exception e) {
+                        logger.error("Unable to exchange authorization for the Access token: " + e.getMessage(), e);
+
+                        // Remove the oidc request cookie
+                        removeOidcRequestCookie(httpServletResponse);
+
+                        // Forward to the error page
+                        forwardToMessagePage(httpServletRequest, httpServletResponse, "Unable to exchange authorization for ID token: " + e.getMessage());
```

Comment:
Given that this and other uses format the same message, it may be worth formatting the string once and reusing it for both the logger and the forwardToMessagePage methods.

Reply:
Fixed.